### ISSUE TITLE
[java] Add support for exclusions to a few rules

### ISSUE THREAD

**Issue by nkavian** [pmd/pmd#3589]
**Is your feature request related to a problem? Please describe.**
Example: I want to enforce a minimum length of 4 for [ShortClassName](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#shortclassname), [ShortMethodName](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#shortmethodname), [ShortVariable](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#shortvariable), but my project has a few exceptions. I do not want:
 - to disable the rule,
 - to lower the minimum,
 - to use `// NOPMD` in so many places

**Describe the solution you'd like**
I'd like support for an `exclusions` parameter similar to [FieldNamingConventions](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#fieldnamingconventions) where exceptions to the rule can be provided.

**Describe alternatives you've considered**
I've lowered my minimum to 2 until this can be addressed.

**Additional context**
PMD is awesome.

**Comment by jsotuyod** [comment]
Are you looking to ignore complete files or to whitelist some specific short names for the whole codebase?

I believe that ignoring complete files could be useful at a base-rule level (and could help to bring PMD into pre-existing codebases), but if you want to whitelist any method called `or` regardless of the class it is in, then this wouldn't suffice.

**Comment by nkavian** [comment]
Correct, specific method names across the code base. Sorry if my description wasn't clear. It was crystal clear in my mind when I wrote it, lol.

**Comment by adamhibbert** [comment]
I'd like to do the same, for instance 'id' as a variable is common in my codebase, but I'd like to set ShortVariable minimum to 3, but have had to reduce it to 2 to get around this.
